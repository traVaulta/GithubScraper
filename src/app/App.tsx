import React from 'react';

function App() {
    return (
        <div className="App">
            <header className="App-header">
                Github Scraper
            </header>
        </div>
    );
}

export default App;
