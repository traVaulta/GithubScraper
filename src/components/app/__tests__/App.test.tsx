import React from 'react';
import {render} from '@testing-library/react';
import * as _ from 'lodash';

import App from '../index';

describe(`${App.name}`, () => {
    test('renders with class', () => {
        const {container, getByText} = render(<App/>);

        const noteElement = getByText('Start browsing repositories - enter user profile in search first...');
        const appElement = _.get(container, 'firstChild');

        expect(noteElement).toBeInTheDocument();
        expect(appElement).toHaveClass('container');
    });

    test('regression', () => {
        const {asFragment} = render(<App/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
