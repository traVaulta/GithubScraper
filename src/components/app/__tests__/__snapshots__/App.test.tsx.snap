// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App regression 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="search-bar"
    >
      <div
        class="search-bar search-bar--grouped-item"
      >
        <input
          aria-label="search-bar--input"
          class="search-bar search-bar--input"
          placeholder="Enter GitHub username here..."
          type="text"
          value=""
        />
      </div>
      <div
        class="search-bar search-bar--grouped-item icon"
      >
        <span>
           
          <i
            class="fa fa-search"
          />
        </span>
      </div>
    </div>
    <div
      class="container"
    >
      Start browsing repositories - enter user profile in search first...
    </div>
  </div>
</DocumentFragment>
`;
